src/main/webapp/script.js

```diff
@@ -136,13 +190,40 @@ function schedule() {
   formContent.append("startDate", startDate.trim());
   formContent.append("endDate", endDate.trim());
   appendDurations(formContent);
+  formData = appendWorkingHours(formContent); // JavaScript seems pass by reference, maybe = is unnecessary, but could increase readability?
+  saveSettings();
   postData("/schedule", new URLSearchParams(formContent).toString())
-      .then(getJsonIfOk)
+      .then(checkResponse)
+      .then(getJson)
       .then(updateView)
       .then(refreshCalendar)
+      .catch(restoreScheduleButton)
       .catch(handleNetworkError);
 }
 
+/**
+ * Restores the schedule button as before clicking it.
+ */
+function restoreScheduleButton() {
+  showScheduleButton();
+  handleEmptySelection();
+}
+
+function appendWorkingHours(formData) {
+  workingHours = extractWorkingHours();
+  for(setting in workingHours){
+    formData.append(setting, workingHours[setting]);
+  }
+}
+
+function extractWorkingHours(){
+  var workingHours = new Object();
```

Comment:
make this a const and initialize as `{}` instead. They're basically the same but the object literal syntax is more commonly used and doesn't need a `new` keyword.
https://stackoverflow.com/questions/4597926/what-is-the-difference-between-new-object-and-object-literal-notation

Reply:
I can't make it const, because then I cannot add new keys to it. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/const#:~:text=The%20const%20declaration%20creates%20a,its%20properties)%20can%20be%20altered.
I could add the keys beforehand, but then there is no flexibility in it anymore.

Reply:
I see. But you can still initialize it as an object literal then instead of `new Object()`.

Reply:
That is right. `const` only means that the object assigned to the identifier is not replaced by anything else, but this object's internals may be changed.
```
const person = new Person();
person.name = 'John';
const x = {};
x[7] = 42;
x.foo = 'bar';
```
these are all valid statements.

To indicate a real constant, which would never change, additionally use uppercase name, e.g. `const ROUNDED_PI = 3.1`.